### ISSUE TITLE
[Bug]: Handle no corresponding user for key 

### ISSUE THREAD

**Issue by krrishdholakia** [BerriAI/litellm#1595]
### What happened?

If key doesn't have a corresponding user in the user's table, any query from the key will fail. 

This is a backwards compatibility issue for older versions of the proxy. 

### Relevant log output

```shell
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/litellm/proxy/proxy_server.py", line 386, in user_api_key_auth
    user_max_budget = user_id_information.max_budget
AttributeError: 'NoneType' object has no attribute 'max_budget'
```

### Twitter / LinkedIn details

cc: @Manouchehri

**Comment by ishaan-jaff** [comment]
Fixed here: https://github.com/BerriAI/litellm/commit/63f18e7163789f69cc045818d8a38735b8613e85 

But I think we also need a longer term fix that moves to using getattr instead

**Comment by ghost** [comment]
closing as fix is now live in v`1.19.2`. Please re-open if the issue persists for you @Manouchehri
